sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -92,31 +94,47 @@ object DecimalType extends AbstractDataType {
 
   override private[sql] def simpleString: String = "decimal"
 
-  val Unlimited: DecimalType = DecimalType(None)
+  val Maximum: DecimalType = DecimalType(Some(PrecisionInfo(MAX_PRECISION, 18)))
+
+  val Default: DecimalType = DecimalType(Some(PrecisionInfo(10, 0)))
+
+  val Unlimited: DecimalType = Maximum  // backward compatibility
 
   private[sql] object Fixed {
```

Comment:
i think you can now just remove Fixed and Expression, and use the unapply directly on DecimalType.

These two are private APIs.